### DISCUSSION TITLE
How to make it work on a relationship aggregated column

### DISCUSSION THREAD

**Discussion by eele94** [malzariey/filament-daterangepicker-filter#54] [Q&A] [answered]
I have a advertisers resource and I want to count the ads for the selected daterange using this daterangerpicker.

I would love to just use this column:
```
                TextColumn::make('ads_count')
                    ->url(fn (Advertiser $record) => ListAds::getFilterUrl('advertiser', $record->id))
                    ->counts('ads')
                    ->sortable()
                    ->summarize([
                        Sum::make('ads_count'),
                    ]),
```
and this filter:
```
            ->filters([
                DateRangeFilter::make('ads.created_at'),
            ])
```

but this doesn't do anything

**Comment by malzariey** [comment, marked as answer]
The filter with only work with a database column, using a relationship will not work. 

I believe you would need to add the ads ["created_at"] column to the query of the table ["ads_created_at"] using join. Or you can use with (relation) eloquent method, but you would need to change the DateRangeFilter Query to use having instead of where.

Good Luck.

**Reply by eele94** [reply]
I updated the callback of the counts function and now it's finally working

I think it's kinda hacky to get the filter data and parse everything this way..
Anything more cleaner is always appreciated.
Btw, thanks for the quick reply.

```
fn ($livewire) => ['ads' => function (Builder $query) use ($livewire) {
            $datesString = data_get($livewire->tableFilters, 'ads.last_shown_date.ads.last_shown_date');
            $dates = explode(' ', $datesString);

            if (count($dates) !== 3) {
                return;
            }

            $from = $dates[0];
            $to = $dates[2];

            $format = 'd/m/Y';

            $from = Carbon::createFromFormat($format, $from)->startOfDay();
            $to = Carbon::createFromFormat($format, $to)->endOfDay();

            $query->whereBetween('last_shown_date', [
                $from,
                $to,
            ]);
        }]
```

**Reply by malzariey** [reply]
You could use the $state to get the selected filter

```
->query(function($query,$state){
                        
                })
```
I will consider injecting the startDate and endDate in the query parameters in the next version.
